pkg/curatedpackages/curatedpackages.go

```diff
@@ -51,6 +52,54 @@ func GetVersionBundle(reader Reader, eksaVersion string, spec *v1alpha1.Cluster)
 	return versionsBundle, nil
 }
 
+func formatYamlLine(space, key, value string) string {
+	if value == "" {
+		return ""
+	}
+	return space + key + ": " + value + "\n"
+}
+
+// GetPackageControllerConfiguration returns the default kubernetes version for a Cluster.
+func GetPackageControllerConfiguration(clusterSpec *v1alpha1.ClusterSpec) (result string, err error) {
+	if clusterSpec == nil {
+		return "", nil
+	}
+
+	if clusterSpec.Packages.Controller != nil {
+		result += "controller:\n"
+		result += formatYamlLine("  ", "digest", clusterSpec.Packages.Controller.Digest)
```

Comment:
Is the goal to return a yaml formatted controller config? Maybe yaml marshaller will come in handy here?

Reply:
I tried taht, but this approach was much easier to understand and test.

Reply:
I have used the yaml marshall's functionality and it was easy to understand and also test. We can sync up offline.